pkg/standalone-utils/test/ProtocolFeeSplitter.test.ts

```diff
@@ -0,0 +1,212 @@
+import { ethers } from 'hardhat';
+import { expect } from 'chai';
+import { Contract } from 'ethers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+import TokenList from '@balancer-labs/v2-helpers/src/models/tokens/TokenList';
+import { deploy, deployedAt } from '@balancer-labs/v2-helpers/src/contract';
+import { actionId } from '@balancer-labs/v2-helpers/src/models/misc/actions';
+import Vault from '@balancer-labs/v2-helpers/src/models/vault/Vault';
+import * as expectEvent from '@balancer-labs/v2-helpers/src/test/expectEvent';
+import { toNormalizedWeights } from '@balancer-labs/balancer-js/src';
+import { fp } from '@balancer-labs/v2-helpers/src/numbers';
+import { ZERO_ADDRESS } from '@balancer-labs/v2-helpers/src/constants';
+import { bn } from '../../../pvt/helpers/src/numbers';
+import { JoinPoolRequest, WeightedPoolEncoder } from '@balancer-labs/balancer-js';
+import { expectEqualWithError } from '@balancer-labs/v2-helpers/src/test/relativeError';
+
+const DELEGATE_OWNER = '0xBA1BA1ba1BA1bA1bA1Ba1BA1ba1BA1bA1ba1ba1B';
+
+const NAME = 'Balancer Pool Token';
+const SYMBOL = 'BPT';
+const POOL_SWAP_FEE_PERCENTAGE = fp(0.01);
+const WEIGHTS = toNormalizedWeights([fp(30), fp(70), fp(5), fp(5)]);
+
+describe('ProtocolFeeSplitter', function () {
+  let vault: Vault;
+  let factory: Contract;
+  let protocolFeesCollector: Contract;
+  let protocolFeeSplitter: Contract;
+  let pool: Contract;
+  let poolNoOwner: Contract;
+  let poolDelegatedOwner: Contract;
+  let admin: SignerWithAddress,
+    poolOwner: SignerWithAddress,
+    treasury: SignerWithAddress,
+    liquidityProvider: SignerWithAddress;
+  let assetManagers: string[];
+  let tokens: TokenList;
+  let poolId: string;
+  let poolNoOwnerId: string;
+  let poolDelegatedOwnerId: string;
+
+  before(async () => {
+    [, admin, poolOwner, liquidityProvider, treasury] = await ethers.getSigners();
+  });
+
+  sharedBeforeEach('deploy vault, protocol fees collector & tokens', async () => {
+    vault = await Vault.create({ admin });
+    protocolFeesCollector = await vault.getFeesCollector();
+    tokens = await TokenList.create(['DAI', 'MKR', 'SNX', 'BAT'], { sorted: true });
+
+    await tokens.mint({ to: liquidityProvider, amount: fp(100000) });
+    await tokens.approve({ from: liquidityProvider, to: vault });
+  });
+
+  sharedBeforeEach('deploy tokens, pools & gives initial liquditiy', async () => {
```

Comment:
Jic, have you checked the test coverage?

Reply:
Coverage is not set up in this repo, so no. But I'm pretty sure we got everything covered

Reply:
Can we just test it locally, it shoudn't be a big deal, just to add solidity-coverage and run npx hardhat coverage

Reply:
![coverage](https://user-images.githubusercontent.com/28648109/185372856-5349f1e0-1ccb-4fa9-99e8-a22784f3995e.png)

Reply:
Gewd